tasks/nfs.yml

```diff
@@ -1,22 +1,24 @@
 ---
 - name: Check if NFS packages are needed
   zfs_facts:
-    name: "{{ root_prefix }}/{{ item.name }}"
+    name: "{{ parent_fs }}/{{ item.name }}"
     properties: "sharenfs"
   ignore_errors: True
   register: no_nfs_datasets
   with_items:
     - "{{ filesystems }}"
-  failed_when: ansible_zfs_datasets[0].sharenfs != "off"
+  changed_when: ansible_zfs_datasets[0].sharenfs != "off"
```

Comment:
zfs_facts returns a list of found datasets. My query affects one filesystem and is not recursive, so the expected entry is at index 0.